test/TestUtils/RuntimeInformationExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.CoreSetup.Test
+{
+    public static class RuntimeInformationExtensions
+    {
+        public static string GetExeExtensionForCurrentOSPlatform()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".exe";
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        public static string GetSharedLibraryExtensionForCurrentPlatform()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".dll";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return ".dylib";
+            }
+            else
+            {
+                return ".so";
+            }
+        }
+
+        public static string GetSharedLibraryPrefixForCurrentPlatform()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return "";
+            }
+            else
+            {
+                return "lib";
+            }
+        }
+
+        public static string GetDefaultNugetDirectoryForCurrentPlatform()
```

Comment:
> It'd be cool if we had some static analysis tests for dead code.

There is for full framework, but static analysis can't help you when you are making a library and you make a public method. How is it supposed to know if the code is 'dead', or if you really intended on shipping a public method here for others to consume?

Just another reason to default your visibility to the most hidden possible. (`private` if only the current class uses it, `internal` if only the current assembly uses it, `public` only if you have to)